packages/medusa/src/services/store.ts

```diff
@@ -0,0 +1,292 @@
+      symbol: currencyObject.symbol,
+      symbol_native: currencyObject.symbol_native,
+      name: currencyObject.name,
+    }
+  }
+
+  /**
+   * Updates a store
+   * @param data - an object with the update values.
+   * @return resolves to the update result.
+   */
+  async update(data: AdminPostStoreReq): Promise<Store> {
+    return await this.atomicPhase_(
+      async (transactionManager: EntityManager) => {
+        const storeRepository = transactionManager.getCustomRepository(
+          this.storeRepository_
+        )
+        const currencyRepository = transactionManager.getCustomRepository(
+          this.currencyRepository_
+        )
+
+        const {
+          metadata,
+          default_currency_code,
+          currencies: storeCurrencies,
+          ...rest
+        } = data
+
+        const store = await this.retrieve({ relations: ["currencies"] })
+        if (metadata) {
+          store.metadata = this.setMetadata_(store.id, metadata)
+        }
+
+        if (storeCurrencies) {
+          const defaultCurr =
+            default_currency_code ?? store.default_currency_code
+          const hasDefCurrency = storeCurrencies.find(
+            (c) => c.toLowerCase() === defaultCurr.toLowerCase()
+          )
+
+          // throw if we are trying to remove a currency from store currently used as default
+          if (!hasDefCurrency) {
+            throw new MedusaError(
+              MedusaError.Types.INVALID_DATA,
+              `You are not allowed to remove default currency from store currencies without replacing it as well`
+            )
+          }
+
+          store.currencies = await Promise.all(
+            storeCurrencies.map(async (curr) => {
+              const currency = await currencyRepository.findOne({
+                where: { code: curr.toLowerCase() },
+              })
+
+              if (!currency) {
+                throw new MedusaError(
+                  MedusaError.Types.INVALID_DATA,
+                  `Invalid currency ${curr}`
+                )
+              }
+
+              return currency
+            })
+          )
+        }
+
+        if (default_currency_code) {
+          const storeCurrCodes = store.currencies.map((c) => c.code)
+          const hasDefCurrency = storeCurrCodes.find(
+            (c) => c === default_currency_code.toLowerCase()
+          )
+
+          // throw if store currencies does not have default currency
+          if (!hasDefCurrency) {
+            throw new MedusaError(
+              MedusaError.Types.INVALID_DATA,
+              `Store does not have currency: ${default_currency_code}`
+            )
+          }
+
+          const curr = await currencyRepository.findOne({
+            code: default_currency_code.toLowerCase(),
+          })
+
+          if (!curr) {
+            throw new MedusaError(
+              MedusaError.Types.INVALID_DATA,
+              `Currency ${default_currency_code} not found`
+            )
+          }
```

Comment:
this shouldn't every be reached since the store will never have the default currency (since it's not allowed). This should either be moved above the previous statement or removed.